arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -82,30 +97,33 @@
 docTypeXhtmlPattern = re.compile(r"^<!(?:DOCTYPE\s+)\s*html(?:PUBLIC\s+)?(?:.*-//W3C//DTD\s+(X?HTML)\s)?.*>$", re.IGNORECASE)
 
 FOOTNOTE_LINK_CHILDREN = {qnLinkLoc, qnLinkFootnoteArc, qnLinkFootnote, qnIXbrl11Footnote}
-PERCENT_TYPE = qname("{http://www.xbrl.org/dtr/type/numeric}num:percentItemType")
+PERCENT_TYPE = qname("{http://www.xbrl.org/dtr/type/numeric}num:percentItemType") #  type: ignore[no-untyped-call]
 IXT_NAMESPACES = {ixtNamespaces["ixt v4"], # only tr4 or newer REC is currently recommended
                   ixtNamespaces["ixt v5"]}
 
-def etreeIterWithDepth(node, depth=0):
+def etreeIterWithDepth(node: ModelObject | _Element, depth: int=0): # type: ignore[no-untyped-def]
     yield (node, depth)
     for child in node.iterchildren():
         for n_d in etreeIterWithDepth(child, depth+1):
             yield n_d
 
-def dislosureSystemTypes(disclosureSystem, *args, **kwargs):
+def dislosureSystemTypes(disclosureSystem: DisclosureSystem, *args: tuple[Any], **kwargs: dict[Any, Any]) -> tuple[tuple[str, str]]:
```

Comment:
Python typing has special handling for `*args` and `**kwargs`.

`(*args: Any, **kwargs: Any)`
is equivalent to
`(*args: tuple[Any, ...], **kwargs: dict[str, Any])`

Reply:
Looks like these are present in the other plugins as well so I created shared types that we can reuse for the other files.